mobile/mysterium/openvpn_connection_setup.go

```diff
@@ -29,22 +29,40 @@ import (
 	"github.com/mysteriumnetwork/node/services/openvpn/session"
 )
 
+type openvpn3SessionFactory func(connection.ConnectOptions) (*openvpn3.Session, error)
+
+var errSessionWrapperNotStarted = errors.New("session wrapper not started")
+
 type sessionWrapper struct {
-	session *openvpn3.Session
+	session        *openvpn3.Session
+	sessionFactory openvpn3SessionFactory
 }
 
-func (wrapper *sessionWrapper) Start() error {
-
+func (wrapper *sessionWrapper) Start(options connection.ConnectOptions) error {
+	session, err := wrapper.sessionFactory(options)
+	if err != nil {
+		return err
+	}
+	wrapper.session = session
 	wrapper.session.Start()
 	return nil
 }
 
 func (wrapper *sessionWrapper) Stop() {
-	wrapper.session.Stop()
+	if wrapper.session != nil {
+		wrapper.session.Stop()
+	}
 }
 
 func (wrapper *sessionWrapper) Wait() error {
-	return wrapper.session.Wait()
+	if wrapper.session != nil {
+		return wrapper.session.Wait()
+	}
+	return errSessionWrapperNotStarted
+}
+
+func (wrapper *sessionWrapper) GetSessionConfig() (connection.SessionCreationConfig, error) {
```

Comment:
GetConfig() ? Session is repeated too much in this context and its meaning is different. connection.SessionCreationConfig -> simply ConsumerConfig? It doesn't configure session itself but provides option to send underlying transport configuration

Reply:
👍 for `ConsumerConfig`